File: api/src/main/java/com/vibepay/domain/Order.java
package com.vibepay.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 주문 도메인 모델
 */
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {
    private Long id;
    private Long memberId;
    private String orderNumber;

    // 상품 정보
    private String productName;
    private Long productPrice;
    private Integer quantity;

    // 금액 정보
    private Long totalAmount;
    private Long pointAmount;
    private Long cardAmount;

    // 주문 상태
    private OrderStatus status;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: fo/src/types/order.ts
// 주문 관련 타입 정의
export interface OrderItem {
  productName: string;
  productPrice: number;
  quantity: number;
}

export interface OrderSummary {
  productAmount: number;
  pointUsed: number;
  finalAmount: number;
}

export interface CreateOrderRequest {
  orderNumber: string;
  productName: string;
  productPrice: number;
  quantity: number;
  paymentMethod: 'CARD' | 'POINT' | 'MIXED';
  pointAmount: number;
  cardAmount: number;
  pgAuthToken?: string;
  pgTid?: string;
  mid?: string;
  price?: string;
  currency?: string;
}

export interface CreateOrderResponse {
  orderId: string;
  status: string;
  message: string;
}

export interface Order {
  id: string;
  orderNumber: string;
  productName: string;
  productPrice: number;
  quantity: number;
  totalAmount: number;
  pointAmount: number;
  cardAmount: number;
  status: 'PENDING' | 'PAID' | 'CANCELLED';
  createdAt: string;
  updatedAt: string;
}

export interface OrderListResponse {
  orders: Order[];
  totalCount: number;
  totalPages: number;
  currentPage: number;
  pageSize: number;
}

export interface PointBalance {
  balance: number;
}

export interface PointDeductRequest {
  amount: number;
  orderId: string;
}

export interface PointDeductResponse {
  success: boolean;
  remainingBalance: number;
  message: string;
}